lms/static/scripts/frontend_apps/components/BookPicker.tsx

```diff
@@ -33,40 +28,37 @@ import ErrorDisplay from './ErrorDisplay';
  * Note: The component logic is consistent about using the terminology of
  * "chapter" for content segments in a VitalSource eBook, but presents this
  * to the user as "location" and other slightly less precise language.
- *
- * @param {BookPickerProps} props
  */
-export default function BookPicker({ onCancel, onSelectBook }) {
+export default function BookPicker({
+  onCancel,
+  onSelectBook,
+}: BookPickerProps) {
   const vsService = useService(VitalSourceService);
 
-  const [chapterList, setChapterList] = useState(
-    /** @type {Chapter[]|null} */ (null)
-  );
-  const [book, setBook] = useState(/** @type {Book|null} */ (null));
-  const [chapter, setChapter] = useState(/** @type {Chapter|null} */ (null));
-  const [step, setStep] = useState(
-    /** @type {'select-book'|'select-chapter'} */ ('select-book')
+  const [chapterList, setChapterList] = useState<Chapter[] | null>(null);
+  const [book, setBook] = useState<Book | null>(null);
+  const [chapter, setChapter] = useState<Chapter | null>(null);
+  const [step, setStep] = useState<'select-book' | 'select-chapter'>(
+    'select-book'
   );
-  const [error, setError] = useState(/** @type {Error|null} */ (null));
+  const [error, setError] = useState<Error | null>(null);
 
-  /** @type {(b: Book) => void} */
-  const confirmBook = useCallback(book => {
+  const confirmBook = useCallback((book: Book) => {
     setBook(book);
     setChapterList(null);
     setStep('select-chapter');
   }, []);
 
-  /** @type {(c: Chapter) => void} */
   const confirmChapter = useCallback(
-    chapter => {
-      onSelectBook(/** @type {Book} */ (book), chapter);
+    (chapter: Chapter) => {
+      onSelectBook(book as Book, chapter);
```

Comment:
```suggestion
      onSelectBook(book!, chapter);
```

The origin cast was a non-null cast.